js/group_email.js

```diff
@@ -0,0 +1,187 @@
+let sa = new SheetsApi();
+sa.handleClientLoad();
+
+let sheetHeaders;
+let criteriaIndex = 1;
+let stepNumber = 1;
+
+function updateSignInStatus(isSignedIn) {
+    if (isSignedIn) {
+        sa.getTableHeaders("UPLS").then(res => {
+            sheetHeaders = sa.parseTableHeaders(res);
+            addCriteria();
+            document.getElementById("criteria").classList.remove("invisible");
+        });
+    } else {
+        sa.handleSignInClick();
+    }
+}
+
+function backStep() {
+    if (stepNumber === 1) {
+        window.location.href = "main_page.html";
+    } else if (stepNumber === 2) {
+        document.getElementById("template").classList.add("invisible");
+        document.getElementById("criteria").classList.remove("invisible");
+        stepNumber = 1;
+    } else {
+        document.getElementById("preview").classList.add("invisible");
+        document.getElementById("template").classList.remove("invisible");
+        stepNumber = 2;
+    }
+}
+
+function nextStep() {
+    if (stepNumber === 1) {
+        document.getElementById("template").classList.remove("invisible");
+        document.getElementById("criteria").classList.add("invisible");
+        stepNumber = 2;
+    } else if (stepNumber === 2) {
+        loadDraft();
+    }
+}
+
+function addCriteria() {
+    let options = "";
+    for (let i = 0; i < sheetHeaders.length; i++) {
+        options += "<option>" + sheetHeaders[i] + "</option>";
+    }
+    let str = "<div id=\"criteria_" + criteriaIndex + "\">" +
+        "<table>" +
+        "<tr>" +
+        "<td><button class=\"close_button\" onclick=\"deleteCriteria(this)\"></button></td>" +
+        "<td><label for=\"select_criteria_" + criteriaIndex + "\">Criteria: </label></td>" +
+        "<td><select id=\"select_criteria_" + criteriaIndex + "\" onchange=\"getVocabulary(this.value," + criteriaIndex + ")\">" + options + "</select></td>" +
```

Comment:
The 'onchange' event only triggers when selecting a different option from the list. So, if the user wants to use STUDENT_ID, then they need to select something else and then go back to STUDENT_ID, which isn't very intuitive.